tests/test_advection.py

```diff
@@ -134,12 +134,13 @@ def SubmergeParticle(particle, fieldset, time):
         particle.depth = 0
         AdvectionRK4(particle, fieldset, time)  # perform a 2D advection because vertical flow will always push up in this case
         particle.time = time + particle.dt  # to not trigger kernels again, otherwise infinite loop
+        particle.succeeded()
 
     recovery_dict = {ErrorCode.ErrorOutOfBounds: DeleteParticle}
     if wErrorThroughSurface:
         recovery_dict[ErrorCode.ErrorThroughSurface] = SubmergeParticle
 
-    pset = ParticleSet(fieldset=fieldset, pclass=ptype[mode], lon=0.5, lat=0.5, depth=0.9)
+    pset = ParticleSet(fieldset=fieldset, pclass=ptype[mode], lon=[0.5, ], lat=[0.5, ], depth=[0.9, ])
```

Comment:
Why is this change necessary? It should be possible to define `lon`, `lat` etc as scalars?

Reply:
actually, yes, that was when I tested the function to find the error, but it can be restored to its original state.